cloud/pkg/devicecontroller/controller/downstream.go

```diff
@@ -831,6 +831,22 @@ func deleteDeviceModelAndVisitors(deviceModel *v1alpha2.DeviceModel, deviceProfi
 	}
 }
 
+// getDeviceNode get the Node bound to this device
+func getDeviceNode(device *v1alpha2.Device) string {
+	nodeID := device.Spec.NodeName
+	if len(nodeID) == 0 {
+		// The useage of device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values[0] is an issue left over from history.
+		// It not really implement 'NodeSelector' feature just like k8s, this problem mentioned at https://github.com/kubeedge/kubeedge/issues/4763
+
+		// TODO: implement NodeSelector feature. When complete, device.Spec.NodeName field means this device onto what node, and device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values[0] can be deleted at here.
```

Comment:
we can delete the field of `NodeSelector` directly and it will be proposed in https://github.com/kubeedge/kubeedge/issues/4763

Reply:
Delete it directly may cause compatibility issues, and I prefer to do this after #4763 completed, what do you think?

Reply:
+1, must keep compatible with prefer version.